src/includes/environment.js

```diff
@@ -749,61 +749,191 @@ pub.bleeds = function(top, right, bottom, left) {
 
 
 /**
- * Prints out all properties and values off an object in a recursive manner to the console. Useful for inspecting (or debugging) nested variable. the default value for the recursion is maxlevel = 2.
+ * @description Inspects a given object or any other data item and prints the result to the console. This is useful for inspecting or debugging any kind of variable or data item. The optional settings object allows to control the function's output. The following parameters can be set in the settings object:
+ * <code>showProps<code>: Show or hide properties. Default: <code>true<code>
+ * <code>showValues<code>: Show or hide values. Default: <code>true<code>
+ * <code>showMethods<code>: Show or hide methods. Default: <code>false<code>
+ * <code>maxLevel<code>: Chooses how many levels of properties should be inspected recursively. Default: <code>1<code>
+ * <code>propList<code>: Allows to pass an array of property names to show. If propList is not set all properties will be shown. Default: <code>[]<code> (no propList)
+ * If no settings object is set, the default values will be used.
  *
  * @cat Output
  * @method inspect
- * @param  {Object} obj The Object to be inspected.
- * @param  {Number} maxlevel Recursion limit, default maxlevel = 2.
+ * @param  {Object} obj An object or any other data item to be inspected.
+ * @param  {Object} settings A settings object to control the function's behavior.
+ *
+ * @example
+ * // inspecting a string
+ * b.inspect("foo");
+ *
+ * @example
+ * // inspecting the current page, its methods and an additional level of properties
+ * b.inspect(b.page(), {showMethods: true, maxLevel: 2})
+ *
+ * @example
+ * // inspecting an ellipse, listing only the properties "geometricBounds" and "strokeWeight"
+ * var myEllipse = b.ellipse(0, 0, 10, 10);
+ * b.inspect(myEllipse, {maxLevel: 2, propList: ["geometricBounds, strokeWeight"]});
  */
-pub.inspect = function(obj, maxlevel, level, propname) {
-  checkNull(obj);
+pub.inspect = function (obj, settings, level, branchArray, branchEnd) {
```

Comment:
They are used internally only for the recursive call of the function, so that's why they should not be exposed to the user (as they are not supposed to use these parameters). To help with reviewing though:

`level` holds the current level (top level being `0`)
`branchArray` is an array of booleans that holds the info how the branches need to be drawn in the current line. Something like `[true, true, true]` renders to

```
|   |   |   
```
 while something like `[true, false, true]` renders to
```
|       |   
```
`branchEnd` is a boolean that holds the information if the parent property is at the very end of a branch. This is needed for correctly drawing the tree.